capstone/src/main/java/com/google/univiz/scorecard/CollegeDataApiImpl.java

```diff
@@ -0,0 +1,39 @@
+package com.google.univiz.scorecard;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.google.univiz.CollegeData;
+import com.google.univiz.CollegeDataApi;
+import com.google.univiz.api.CollegeId;
+import com.ryanharter.auto.value.gson.GenerateTypeAdapter;
+import java.util.ArrayList;
+import java.util.List;
+import javax.inject.Inject;
+
+public class CollegeDataApiImpl implements CollegeDataApi {
+  private final ScorecardConverter scorecardConverter;
+  private final CollegeIdReaderProvider readerProvider;
+
+  @Inject
+  public CollegeDataApiImpl(
+      ScorecardConverter scorecardConverter, CollegeIdReaderProvider readerProvider) {
+    this.scorecardConverter = scorecardConverter;
+    this.readerProvider = readerProvider;
+  }
+
+  @Override
+  public List<CollegeData> getCollegesById(List<CollegeId> ids) {
+    List<CollegeData> colleges = new ArrayList<>();
+    for (CollegeId id : ids) {
```

Comment:
Philosophical question - they way this is configured we're going to end up sending a REST request for every college id in serial. This might slow things down a bit. 

Can you think of a way to achieve the same result but with a single request?

Reply:
Ah that's a good point, I think I can just build up the url to include all the ids, and then make one REST call, then do deserialization/conversion/the like. I shall do that instead of this.